src/monitoring_tests/fee_test.py

```diff
@@ -0,0 +1,160 @@
+"""
+Fee Test
+"""
+
+from src.monitoring_tests.base_test import BaseTest
+from src.apis.web3api import Web3API
+from src.apis.orderbookapi import OrderbookAPI
+from src.helper_functions import get_logger
+from src.constants import (
+    FEE_ABSOLUTE_DEVIATION_ETH_FLAG,
+    FEE_RELATIVE_DEVIATION_FLAG,
+)
+
+
+class FeeTest(BaseTest):
+    """
+    Class for testing fees.
+    """
+
+    def __init__(self):
+        self.web3_api = Web3API()
+        self.orderbook_api = OrderbookAPI()
+        self.logger = get_logger()
+
+    def run(self, tx_hash) -> bool:  # pylint: disable=too-many-locals
+        """
+        Given a transaction hash, check if there is a partially-fillable order in the settlement.
+        If this is the case, perform multiple tests on the execution of those orders to check if
+        the fee was set correctly.
+        TODO: add more explanations
+        """
+        # get settlement and trades via api
+        transaction = self.web3_api.get_transaction(tx_hash)
+        if transaction is None:
+            return False
+        settlement = self.web3_api.get_settlement(transaction)
+        trades = self.web3_api.get_trades(settlement)
```

Comment:
could web3_api have a method like `get_all` that returns an object like 

```
@dataclass
class SettlementTransaction:
    transaction: TxReceipt
    settlement: SettlementEvent?
    trades: list[Trade]
```

Then this block is `settlement_data = self.web3_api.get_all()` reducing 3 local variables to 1 more compact instance.


In fact, then a large chuck of what happens next is handling partially fillable orders. the class could have a method like `handle_partially_fillable_orders`. I believe just these two suggestions alone would eliminate the need for pylint-ignore here.

Reply:
For now I did not combine all data fetching into a single call on the web3_api side. The main reason is that (not necessarily now but potentially in the future) different tests might fetch different data from this api. For example, the main loop might fetch hashes, but the fee test fetches transactions as well.
Maybe the different tests themselves should implement all the data fetching in their own helper functions?